### PR TITLE
WIP Fix lingering issues with Repository tests

### PR DESCRIPTION
**Pull request by cbj4074**
All I'm looking to do at this point is eliminate calls to the factory `create()` method, in favor of using `make()` and passing the array into the *repository's* own `create()` method.

### CODE REVIEW

File: tests/Feature/Repositories/AccountRepositoryTest.php [hunk 1]
```diff
@@ -39,6 +42,28 @@ public function setRepository()
...
     }
 
+    /**
+     * Make a new User object.
+     *
+     * @param array $userProperties
+     * @param array $accountProperties
+     * @return \App\User
+     */
+    public function makeUser(array $userProperties = [], array $accountProperties = [])
```
Comment (poppabear8883) [inline tests/Feature/Repositories/AccountRepositoryTest.php]: I'm wondering if we can abstract this to the `RepositoryCrudTestCase` and pass in `$this->user` to make this available in all repository tests ?

Can we see multiple tests needing access to similar logic ?
Can we see creating maybe a Trait `use RepositoryTestFactories;` (or something similar) ?

Comment (cbj4074) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Yes, definitely. I'm confident that anything useful in this respect will present itself as I go through the tests within the scope of this PR. I'll identify those traits and pencil them in as I go.

**Comment by poppabear8883** [comment]
Looks good to me

File: tests/Feature/Repositories/AccountRepositoryTest.php (hunk 1 above)
Comment (poppabear8883) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Through further investigation, maybe its better to just *NOT* try to abstract this logic.

It becomes more complicated then what its worth when there isn't any real reason to not just do it this way.

This is the solution that i came up with *without a DTO object*

**RepositoryReadOnlyTestCase**
```php
/**
     * Make a collection of Model instances
     *
     * @param array $classes
     * @return array
     */
    public function make(array $classes)
    {
        $models = collect([]);

        foreach ($classes as $class => $properties) {
            $models->push(factory($class)->make($properties));
        }

        return $models;
    }
```

**AccountRepositoryTest**
```php
/**
     * Make a new User object.
     *
     * @param array $userProperties
     * @param array $accountProperties
     * @return \App\User
     */
    public function makeUser(array $userProperties = [], array $accountProperties = [])
    {
        $models = $this->make([
            $this->user->class() => $userProperties,
            $this->repo->class() => $accountProperties
        ]);

        $user = $models[$this->user->class()];
        $account = $models[$this->repo->class()];

        $user = $this->user->create([
            'username' => $user->username,
            'password' => $user->password,
            'account' => $account->toArray(),
        ]);

        return $user;
    }
```
As you can see here its not doing much for us. Even creating a DTO isn't going to make any difference.

Comment (cbj4074) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Okay, having read your review, I agree. I'll revert my previous commit and then go forward with the examples your provided in your previous reply.

Comment (poppabear8883) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Previous reply ?

Comment (cbj4074) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Yeah, the one I was replying to, in which you proposed a solution: https://github.com/indiehd/web-api/pull/93#discussion_r219548704

Comment (poppabear8883) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: No No, i wasn't proposing that solution, that solution is what i was using as an example as to how much effort would be involved into abstracting this logic where you get the same results from the one-liners used below.

I am suggesting no abstraction, and suggesting when you need to "make" a new model instance simply just call the `factory...->make()` method as you have been doing.

Comment (cbj4074) [reply tests/Feature/Repositories/AccountRepositoryTest.php]: Ahhh, I'm with you now. Sorry... I was focusing on the rationale and not the actual code. Understood.

**Pull request merged**